plugins/inputs/bond/bond.go

```diff
@@ -118,10 +159,112 @@ func (bond *Bond) gatherBondPart(bondName string, rawFile string, acc telegraf.A
 	return fmt.Errorf("Couldn't find status info for '%s' ", bondName)
 }
 
+func (bond *Bond) readSysFiles(bondDir string) (sysFiles, error) {
+	/*
+		Files we may need
+		bonding/mode
+		bonding/slaves
+		bonding/ad_num_ports
+
+		We load files here first to allow for easier testing
+	*/
+	var output sysFiles
+
+	file, err := os.ReadFile(bondDir + "/bonding/mode")
+	if err != nil {
+		return sysFiles{}, fmt.Errorf("error inspecting '%q' interface: %v", bondDir+"/bonding/mode", err)
+	}
+	output.ModeFile = strings.TrimSpace(string(file))
+	file, err = os.ReadFile(bondDir + "/bonding/slaves")
+	if err != nil {
+		return sysFiles{}, fmt.Errorf("error inspecting '%q' interface: %v", bondDir+"/bonding/slaves", err)
+	}
+	output.SlaveFile = strings.TrimSpace(string(file))
+	if bond.BondType == "IEEE 802.3ad Dynamic link aggregation" {
+		file, err = os.ReadFile(bondDir + "/bonding/ad_num_ports")
+		if err != nil {
+			return sysFiles{}, fmt.Errorf("error inspecting '%q' interface: %v", bondDir+"/bonding/ad_num_ports", err)
+		}
+		output.ADPortsFile = strings.TrimSpace(string(file))
+	}
+	return output, nil
+}
+
+func (bond *Bond) gatherSysDetails(bondName string, files sysFiles, acc telegraf.Accumulator) {
+	var mode string
+	var slaves []string
+	var slavesTmp []string
+	var adPortCount int
+
+	// To start with, we get the bond operating mode
+	scanner := bufio.NewScanner(strings.NewReader(files.ModeFile))
+	/*
+		This file should only have one line in it
+		so we can simply scan the one line
+	*/
+	for scanner.Scan() {
+		line := scanner.Text()
+		mode = strings.TrimSpace(strings.Split(line, " ")[0])
+	}
```

Comment:
Ok, but isn't this a bit overshooting? How about
```suggestion
	// The file has only one line, so read it and split.
	line, err := os.ReadFile(files.ModeFile)
	if err != nil {
		acc.AddError(err)
		return
	}
	mode := strings.SplitN(string(line), " ", 2)[0]
	mode = strings.TrimSpace(mode)
```

Reply:
I'm really just following the same pattern throughout the plugin. And following that pattern made adding testing for the `/sys` stuff a bit easier, so...